emissionsapi/preprocess.py

```diff
@@ -97,12 +96,14 @@ def read_file(ncfile):
 
     # Get time reference from the meta data.
     # Seems like there are named differently in the different gdal versions.
-    time_reference = iso8601.parse_date(
+    time_reference = dateutil.parser.parse(
         meta_data.get('NC_GLOBAL#time_reference') or
         meta_data['time_reference'])
     timestamps = []
     for dt in deltatime:
-        timestamps.append(time_reference + timedelta(milliseconds=dt.item()))
+        timestamps.append(
+            time_reference + relativedelta(microseconds=dt.item())
```

Comment:
Ups, you are right, @shaardie, this should be milliseconds not microseconds. Otherwise, we end up with slightly incorrect dates:

```sh
% gdalinfo data/S5P_OFFL_L2__CO_____20190910T094724_20190910T112854_09889_01_010302_20190916T091418.nc | grep delta_time
  PRODUCT_delta_time_long_name=offset of start time of measurement relative to time_reference
  PRODUCT_delta_time_units=milliseconds since 2019-09-10 00:00:00 ← that's the reference time
```
But since we are dealing with simple time deltas, I don't think we need `relativedelta` here at all. Just keeping the `timedelta` from Python's standard library should work just fine:

```suggestion
            time_reference + timedelta(milliseconds=dt.item())
```

Reply:
Actually, I did make this change, but I committed the suggestion. And then, I accidentally rebased `upstream/master` over it. 

I think it makes sense to use
```time_reference + relativedelta(microseconds=1e3*dt.item())``` in order to unify the Python package for timestamps.

Reply:
I would be happy with `time_reference + relativedelta(microseconds=1e3*dt.item())`. If you put it back, I would merge it.

Reply:
> I think it makes sense to use
time_reference + relativedelta(microseconds=1e3*dt.item()) in order to unify the Python package for timestamps.

I don't really mind having this merged with `relativedelta` but it's not more unified than using `timedelta` since dateutils are basically just an extension of the standard libraries capabilities (e.g. the type of `time_reference` already is `datetime`). So you are using them if you want to or not. `relativetime` compared to `timedelta` just has some extended capabilities (which are unnecessary here).

Anyway, decide on the path forward yourself ;-)

Reply:
> time_reference + relativedelta(microseconds=1e3*dt.item())
I chose this solution. 👍